site/documentation/content/api/device-connection/index.md

```diff
@@ -107,6 +107,176 @@ Implementors of this API may return a *404* status code in order to indicate tha
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on a message to remove the mapping information regarding the command-handling protocol adapter instance for a device in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name                  | Mandatory | Location                 | AMQP Type | Description |
+| :-------------------- | :-------: | :----------------------- | :-------- | :---------- |
+| *subject*             | yes       | *properties*             | *string*  | MUST be set to `remove-cmd-handling-adapter-instance`. |
+| *adapter_instance_id* | yes       | *application-properties* | *string*  | The identifier of the protocol adapter instance to remove the mapping information for. Only if this adapter instance is currently associated with the device or gateway identified by the *device_id* property, the mapping entry will be removed. |
+
+The body of the message SHOULD be empty and will be ignored if it is not.
+
+**Response Message Format**
+
+A response to a *set command-handling adapter instance for device* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's *status* property may contain the following codes:
+
+| Code  | Description |
+| :---- | :---------- |
+| *204* | OK, the adapter instance mapping information for the device has been removed. |
+| *400* | Bad Request, the command-handling adapter adapter instance information has not been updated due to there being no such mapping entry or due to missing data in the request. |
+
+Implementors of this API may return a *404* status code in order to indicate that no device with the given identifier exists for the given tenant or that no adapter instance is associated with the device. Furthermore, in case the currently mapped adapter instance doesn't match the given adapter instance id, a *412* status code may be returned. However, performing any of these checks is optional.
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+## Get command-handling protocol adapter instances for device
+
+Clients use this command to *get* information about the adapter instances that can handle command & control messages for the given device, possibly using one of the given gateways.
+
+As part of this request, the client is supposed to provide the list of gateways that may act on behalf of the given device. This API doesn't mandate checks on the validity of the given device and the gateway list in order not to introduce a dependency on the *Device Registration API*.
```

Comment:
As part of *the* request message ...?

IMHO a hint would be useful as to where the client might get the *list of gateways that may act on behalf of the given device* ...